### ISSUE TITLE
Swing angle automation resets temperature

### ISSUE THREAD

**Issue by Byter09** [mill1000/midea-ac-py#121]
Hello again!

I've been playing around with automation after #118 wasn't a real issue, and I've noticed a weird behaviour. I have the following automation:

```yaml
alias: Klimaanlage
description: ""
trigger:
  - platform: state
    entity_id:
      - climate.btr_ac
    attribute: current_temperature
    for:
      hours: 0
      minutes: 3
      seconds: 0
  - platform: state
    entity_id:
      - climate.btr_ac
    attribute: temperature
    for:
      hours: 0
      minutes: 0
      seconds: 10
condition: []
action:
  - if:
      - condition: numeric_state
        entity_id: climate.btr_ac
        attribute: temperature
        below: sensor.btr_ac_indoor_temperature
    then:
      - device_id: ea73cdbd9092d9ba3f85d2391ba4ca2c
        domain: climate
        entity_id: 45946b956601e3ce6f2bab2ce2a5fdc5
        type: set_hvac_mode
        hvac_mode: auto
        enabled: true
      - device_id: ea73cdbd9092d9ba3f85d2391ba4ca2c
        domain: select
        entity_id: 83aae9c863bff5fe212b5b3bd2772612
        type: select_option
        option: pos_5
    else:
      - device_id: ea73cdbd9092d9ba3f85d2391ba4ca2c
        domain: climate
        entity_id: 45946b956601e3ce6f2bab2ce2a5fdc5
        type: set_hvac_mode
        hvac_mode: "off"
        enabled: true
    enabled: true
mode: restart
```

Basically, when the temperature is above the target temperature, turn on the AC in auto mode, otherwise turn it off.

When turning it on, I also set the swing angle. This is specific to my setup and where the AC is in my room, and is quite important to have it work effectively.

However, I noticed the AC turning off when I set the target temperature to 20°C. I was wondering why, so I took apart the automation and tested each action separately. The last one I tried was the swing angle setting because I couldn't believe it at first. But after several tries, and various ways to set the angle, no matter what I did, it always reset the temperature to 21°C, which caused the automation to re-trigger and turn off the whole AC again.

I am quite unsure what's going on there and I'm not sure this is just some weird HA quirk again. If you need anything to fix this or need me to try something, don't hesitate to ask.

Again, thanks for this wonderful integration 😄

**Comment by Byter09** [comment]
Okay after several restarts and enabling the debug log, the problem stopped occurring. 

Before the restarts I gave it a try by disabling the automation and then only using the angle controls in the device settings. This also reset the temperature to 21°C.

I captured this here, but the file is massive because I didn't restart yet I presume.

[error_log-1.txt](https://github.com/mill1000/midea-ac-py/files/15177075/error_log-1.txt)


Then, after the restart I tried again and this time the temperature stays put where I set it. Even the automation works again... It's like I'm attracting heisenbugs. Got similar things happening at work 😅

Anyway, thanks @v1k70rk4 for the tips. I've been using climate related automations for a while and I prefer having them not split up into multiples that I then have to correlate and check traces on what happened, should they somehow interfere with each other. If a sequence of actions gets too long I also use scripts but that is rarely needed. Also the temperature information I'm using to decide on the AC mode is using the internal temperature of the AC. It is at the air-intake so it should be accurate enough. I am going to use a temperature probe at the other end of a second room though so it's active when that room gets hot. And yeah, I know Auto is not optimal, but my AC goes into fan only mode if the temperature is 1°C above the target and only above that starts cooling. This is perfect because it mixes the cold air from one room with the warm air of another, before pulling nearly 1200W instead of the fan-only 13W 😅

Basically, I'm back to square one and can't reproduce the issue myself now. Guess I'll see whether it happens in the future again.

I don't know what else could cause this. I only have a single automation setting anything on the AC. I have another one but that only sends me a notification if the filter alert goes off. I'll see what happens and keep the target at 20 and if it ever goes to 21 again I'll know that the bug happened again.

Maybe this is also some kind of HA issue itself but I don't know enough about HAs internals to be sure.


Anyway, I'll leave this issue open I guess but if you want to close it due to not even me being able to reproduce it now, go ahead. I can reopen if I know more I guess.

**Comment by mill1000** [comment]
> Before the restarts I gave it a try by disabling the automation and then only using the angle controls in the device settings. This also reset the temperature to 21°C.

I was about to say it could a device quirk...but not if restarting HA made it go away.

I'll close this for now. Feel free to re-open if the issue comes up again.
